src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -673,7 +586,7 @@ IN_PROCESS_APPLICATION::CreateHandler(
     HRESULT hr = S_OK;
     IREQUEST_HANDLER* pHandler = NULL;
 
-    pHandler = new IN_PROCESS_HANDLER(pHttpContext, this);
+    pHandler = new IN_PROCESS_HANDLER(CreateReferencedApplication(this), pHttpContext, m_RequestHandler, m_RequestHandlerContext, m_AsyncCompletionHandler);
```

Comment:
Rename `CreateReferencedApplication` to something else as you aren't creating the application.